lang/cs/Org.Apache.REEF.Common/metrics/Api/IImmutableMetric.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+namespace Org.Apache.REEF.Common.Metrics.Api
+{
+    /// <summary>
+    /// Immutable metric interface. All the metrics put in the <see cref="IMetricsRecord"/> 
+    /// by <see cref="IMetricsSource"/> are kept as <see cref="IImmutableMetric"/>. This interface
+    /// simply is a combination of functions in <see cref="IImmutableMetricCommonFunctions"/> and 
+    /// <see cref="IImmutableMetricSpecificFunctions"/>.
+    /// </summary>
+    public interface IImmutableMetric : IImmutableMetricCommonFunctions, IImmutableMetricSpecificFunctions
```

Comment:
This is the immutable record of the metric taken during snapshot and hence has a timestamp.